platform/mbed_itm.h

```diff
@@ -0,0 +1,58 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if defined(DEVICE_ITM)
+
+#include "hal/itm_api.h"
+
+class SingleWireOutput : public FileHandle {
+public:
+    SingleWireOutput(void) {
+        itm_init();
+    }
+
+    virtual ssize_t write(const void *buffer, size_t size) {
+        const unsigned char *buf = static_cast<const unsigned char *>(buffer);
+        for (size_t i = 0; i < size; i++) {
+            ITM_SendChar(buf[i]);
```

Comment:
This isn't in itm_api.h? Skipping a layer?

Reply:
ITM_SendChar is defined in:

```
cmsis/TARGET_CORTEX_M/core_cm3.h
cmsis/TARGET_CORTEX_M/core_cm4.h
etc.
```

I'm not sure how to explicitly include the right header file based on core.

Reply:
ah, `#include "cmsis.h"` should do the trick.

Reply:
yes, cmsis should be used as a generic cmsis header file